### ISSUE TITLE
Language server protocol support

### ISSUE THREAD

**Issue by rombert** [emmetio/emmet#597]
I was looking at Emmet these days and trying to find a way to integrate it into Eclipse. The most straightforward way seems to be to implement a [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) implementation that can then be reused in multiple IDEs, including Eclipse.

The Language Server Protocol has a large list of supported tools - see https://microsoft.github.io/language-server-protocol/implementors/tools/ , and implementing it would potentially fix #215  as well.

**Comment by sergeche** [comment]
Yes, LSP would be the best solution here, but for now it requires too much effort for me to implement 😬 Might not have enough time for this

**Comment by windwp** [comment]
https://github.com/windwp/emmet-ls
Hi i fork a repo of another guy and try to fix some issue to create a lsp of emmet.
how can i get a correct cursor position after emmet expand.

**Comment by sergeche** [comment]
Emmet produces snippets with caret placeholders, produced by `output.field` option. See https://github.com/emmetio/emmet/blob/master/src/config.ts#L157 for more info

**Comment by windwp** [comment]
@sergeche Thank it work.
